File: backend/src/chat/chat-admin.controller.ts
import { Body, Controller, Delete, Get, Param, Patch, Post, Req, UseGuards } from '@nestjs/common';
import { Request } from 'express';
import { AuthGuard } from '../common/guards/auth.guard';
import { TenantScopeGuard } from '../common/guards/tenant-scope.guard';
import { Roles } from '../common/decorators/roles.decorator';
import { RolesGuard } from '../common/guards/roles.guard';
import { ChatService } from './chat.service';
import { CreateChatUserDto } from './dto/create-chat-user.dto';
import { UpdateChatUserDto } from './dto/update-chat-user.dto';

@Controller('tenants/:tenantId/chat')
@UseGuards(AuthGuard, TenantScopeGuard, RolesGuard)
@Roles('admin', 'tenant')
export class ChatAdminController {
  constructor(private readonly chatService: ChatService) {}

  @Get('users')
  listUsers(@Req() request: Request & { auth?: any }, @Param('tenantId') tenantId: string) {
    return this.chatService.adminListUsers(tenantId);
  }

  @Post('users')
  createUser(
    @Req() request: Request & { auth?: any },
    @Param('tenantId') tenantId: string,
    @Body() dto: CreateChatUserDto
  ) {
    return this.chatService.adminCreateUser(tenantId, dto);
  }

  @Patch('users/:id')
  updateUser(
    @Req() request: Request & { auth?: any },
    @Param('tenantId') tenantId: string,
    @Param('id') id: string,
    @Body() dto: UpdateChatUserDto
  ) {
    return this.chatService.adminUpdateUser(tenantId, id, dto);
  }

  @Delete('users/:id')
  deleteUser(
    @Req() request: Request & { auth?: any },
    @Param('tenantId') tenantId: string,
    @Param('id') id: string
  ) {
    return this.chatService.adminDeleteUser(tenantId, id);
  }

  @Get('conversations')
  listConversations(
    @Req() request: Request & { auth?: any },
    @Param('tenantId') tenantId: string
  ) {
    return this.chatService.adminListConversations(tenantId);
  }

  @Get('conversations/:id/messages')
  listMessages(
    @Req() request: Request & { auth?: any },
    @Param('tenantId') tenantId: string,
    @Param('id') id: string
  ) {
    return this.chatService.adminListMessages(tenantId, id);
  }

  @Delete('conversations/:id')
  deleteConversation(
    @Req() request: Request & { auth?: any },
    @Param('tenantId') tenantId: string,
    @Param('id') id: string
  ) {
    return this.chatService.adminDeleteConversation(tenantId, id);
  }
}

File: backend-java/src/main/java/com/neria/manager/chat/ChatAdminController.java
package com.neria.manager.chat;

import com.neria.manager.common.security.AuthContext;
import com.neria.manager.common.security.AuthUtils;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/tenants/{tenantId}/chat")
public class ChatAdminController {
  private final ChatService chatService;

  public ChatAdminController(ChatService chatService) {
    this.chatService = chatService;
  }

  private void requireTenantScope(HttpServletRequest request, String tenantId) {
    AuthContext auth = AuthUtils.requireAuth(request);
    AuthUtils.requireRole(auth, "admin", "tenant");
    AuthUtils.requireTenantScope(auth, tenantId);
  }

  @GetMapping("/users")
  public Object listUsers(HttpServletRequest request, @PathVariable String tenantId) {
    requireTenantScope(request, tenantId);
    return chatService.adminListUsers(tenantId);
  }

  @PostMapping("/users")
  public Object createUser(
      HttpServletRequest request,
      @PathVariable String tenantId,
      @RequestBody ChatService.CreateChatUserRequest dto) {
    requireTenantScope(request, tenantId);
    return chatService.adminCreateUser(tenantId, dto);
  }

  @PatchMapping("/users/{id}")
  public Object updateUser(
      HttpServletRequest request,
      @PathVariable String tenantId,
      @PathVariable String id,
      @RequestBody ChatService.UpdateChatUserRequest dto) {
    requireTenantScope(request, tenantId);
    return chatService.adminUpdateUser(tenantId, id, dto);
  }

  @DeleteMapping("/users/{id}")
  public Object deleteUser(
      HttpServletRequest request, @PathVariable String tenantId, @PathVariable String id) {
    requireTenantScope(request, tenantId);
    return chatService.adminDeleteUser(tenantId, id);
  }

  @GetMapping("/conversations")
  public Object listConversations(HttpServletRequest request, @PathVariable String tenantId) {
    requireTenantScope(request, tenantId);
    return chatService.adminListConversations(tenantId);
  }

  @GetMapping("/conversations/{id}/messages")
  public Object listMessages(
      HttpServletRequest request, @PathVariable String tenantId, @PathVariable String id) {
    requireTenantScope(request, tenantId);
    return chatService.adminListMessages(tenantId, id);
  }

  @DeleteMapping("/conversations/{id}")
  public Object deleteConversation(
      HttpServletRequest request, @PathVariable String tenantId, @PathVariable String id) {
    requireTenantScope(request, tenantId);
    return chatService.adminDeleteConversation(tenantId, id);
  }
}
